dlt/destinations/dataset/ibis_relation.py

```diff
@@ -173,6 +214,32 @@ def _get_filtered_columns_schema(self, columns: Sequence[str]) -> TTableSchemaCo
             # here we just break the column schema inheritance chain
             return None
 
+    def _join_to_root_table(self) -> "ReadableIbisRelation":
+        """Join the current table to the root table. If the current table is root, it's a no-op."""
+        LOAD_ID_COL = "_dlt_load_id"
+
+        table_schema = self.schema.tables[self.table_name]
+        if is_root_table(table_schema):
+            return self
+
+        root_key = get_root_key(table_schema)
+        # TODO setup another case that traverse parent-row keys to join nested tables without root_key
+        if root_key is None:
+            raise KeyError(
+                "ReadableIbisRelation requires a `root_key` hint to join non-root tables. "
```

Comment:
we should add a link to the transformation docs in the future for this exception.